e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -145,14 +146,89 @@ public async Task Method_ServiceInvokeDeviceMethodWithUnknownDeviceThrows()
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponse_Mqtt()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Mqtt_Tcp_Only, SetModuleReceiveMethodAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponse_MqttWs()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Mqtt_WebSocket_Only, SetModuleReceiveMethodAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponseWithDefaultMethodHandler_Mqtt()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Mqtt_Tcp_Only, SetModuleReceiveMethodDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponseWithDefaultMethodHandler_MqttWs()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Mqtt_WebSocket_Only, SetModuleReceiveMethodDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponse_Amqp()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Amqp_Tcp_Only, SetModuleReceiveMethodAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponse_AmqpWs()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Amqp_WebSocket_Only, SetModuleReceiveMethodAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponseWithDefaultMethodHandler_Amqp()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Amqp_Tcp_Only, SetModuleReceiveMethodDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ModuleReceivesMethodAndResponseWithDefaultMethodHandler_AmqpWs()
+        {
+            await SendMethodAndRespondAsync(Client.TransportType.Amqp_WebSocket_Only, SetModuleReceiveMethodDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        [LoggedTestMethod]
+        public async Task Method_ServiceInvokeDeviceMethodWithUnknownModuleThrows()
+        {
+            // setup
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, "ModuleNotFoundTest").ConfigureAwait(false);
+            using var serviceClient = ServiceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            var methodInvocation = new CloudToDeviceMethod("SetTelemetryInterval");
+            methodInvocation.SetPayloadJson("10");
+
+            // act
+            ErrorCode actualErrorCode = ErrorCode.InvalidErrorCode;
```

Comment:
why do we need to initialize this to InvalidErrorCode?

Reply:
So that the value of the field can be compared outside of the try/catch block.

Reply:
If the catch doesn't execute, then the comparison will fail